### PR TITLE
Feature/validate transitions

### PR DESCRIPTION
**Pull request by seb231**
Model now checks to see setting to setting transitions are valid

### CODE REVIEW

File: src/cljc/witan/send/params.cljc [hunk 1]
```diff
@@ -340,10 +342,15 @@
...
+                    valid-trans (get valid-transitions setting)
                     obs (get-in observations [ay state])
                     ay-obs (get observations-per-ay ay)
-                    settings (get valid-settings [ay need])
-                    prior (->> (zipmap (vec settings) (repeat (/ 1.0 (count settings))))
+                    settings (->> (get valid-settings [ay need])
+                                  vec
+                                  (into valid-trans)
+                                  u/duplicates
+                                  vec)
```
Comment (otfrom) [inline src/cljc/witan/send/params.cljc]: what is going on here? I'm not sure why there are 2 calls to `vec` and what `duplicates` is supposed to be doing.

File: src/cljc/witan/send/params.cljc [hunk 2]
```diff
@@ -300,16 +300,18 @@
                                {} observations)
         prior-per-year (continue-for-latter-ays prior-per-year academic-years)]
     (reduce (fn [coll [ay state]]
-              (if-let [beta-params (merge-with +
-                                               (get-in observations [ay state])
-                                               (get prior-per-year ay))]
-                (assoc coll [ay state] beta-params)
+              (if (< 1 (count (get valid-transitions (second (s/need-setting state)))))
```
Comment (otfrom) [inline src/cljc/witan/send/params.cljc]: Can we turn this into a predicate that says what it means?

File: src/cljc/witan/send/params.cljc (hunk 1 above)
Comment (seb231) [reply src/cljc/witan/send/params.cljc]: Getting the valid-settings and valid-trans into the same format then combining them, and removing the duplicates, then putting it back into the format the next part of the process expects (a vector)

Comment (otfrom) [reply src/cljc/witan/send/params.cljc]: you can remove actual duplicates by doing `(into #{} <your vector>)`

File: src/cljc/witan/send/params.cljc [hunk 3]
```diff
@@ -300,16 +303,18 @@
                                {} observations)
         prior-per-year (continue-for-latter-ays prior-per-year academic-years)]
     (reduce (fn [coll [ay state]]
-              (if-let [beta-params (merge-with +
-                                               (get-in observations [ay state])
-                                               (get prior-per-year ay))]
-                (assoc coll [ay state] beta-params)
+              (if (any-valid-transitions? (count (get valid-transitions (second (s/need-setting state)))))
```
Comment (otfrom) [inline src/cljc/witan/send/params.cljc]: I was thinking the call here would look like `(if (any-valid-transition? valid-transitions state)`

File: src/cljc/witan/send/params.cljc (hunk 1 above)
Comment (seb231) [reply src/cljc/witan/send/params.cljc]: Whoops not removing duplicates, keeping them!

**Pull request merged**